### DISCUSSION TITLE
Private exit node?

### DISCUSSION THREAD

**Discussion by beardstack** [oxen-io/lokinet#2042] [Q&A] [answered]
How can someone run a private exit node?

**Comment by majestrate** [comment, marked as answer]
our docs are still kind of sparse but docs on setting up an exit can be found here: 
https://github.com/oxen-io/lokinet/blob/dev/docs/exit-setup.md

to make it private you can either not share the .loki address and/or require an auth code in order to access.

to make your .loki address require an auth code use the following lokinet config snippet:

```
[network]
auth-type=file
auth-file=auth.txt
```

the file at `/var/lib/lokinet/auth.txt` will be checked for hashed passwords each time you try to authenticate with the .loki address. (make sure that file is readable by the `_lokinet` user)

passwords are using standard format done by a tool in `apache2-utils` called `mkpasswd` which will prompt you for a password and put out a secure hashed and salted value to put into that auth.txt file. you can have as many hashed passwords in that auth.txt as you want, and they each go on their own separate line, no trailing spaces.

**Comment by beardstack** [comment]
Thanks!
